content/docs/reference/how.md

```diff
@@ -7,44 +7,89 @@ menu:
+
+1. A language executor, which is a binary named `pulumi-language-<language-name>`, which Pulumi uses to the language your program is written in (e.g. Node or Python). This binary is distributed with the Pulumi CLI.
+2. A language runtime, which is responsible for preparing your program to be executed and observes its execution in order to detect resource registrations.  When a resource is _registered_ (via `new Resource()` in JavaScript or `Resource(...)` in Python) the language runtime communicates this request back to the _deployment engine_. The language runtime is distributed as a regular package, just like any other code you might depend on it your program.  For example, the Node runtime is contained in the `@pulumi/pulumi` package published to NPM and the Python runtime is contained in the `pulumi` package published to PyPI.
+
+## Deployment Engine
+
+The _deployment engine_ is responsible for computing the set of operations needed to drive the current state of your infrastructure into the desired state expressed by your program.  When a _resource registration_ is received from the language host, the engine consults the existing state to determine if that resource has been created before. If it has not, the engine uses a _resource provider_ in order to create it.  If it already exists, the engine works with the resource provider to determine what, if anything, has changed by comparing the old state of the resource with the new desired state of the resource as expressed by the program. If there are changes, the engine determines if it can _update_ the resource in place or if it must _replace_ it by _creating_ a new version and _deleting_ the old version. The decision depends on what properties of the resource are changing and the type of the resource itself.  When the language host communicates to the engine that it has completed execution of the Pulumi program, the engine looks for any existing resources that exist but it did not see a new resource registration for and schedules these resources for deletion.
+
+The deployment engine is embedded into the `pulumi` CLI itself.
+
+## Resource Providers
+
+A resource provider is made up of two different pieces:
+
+1. A _resource plugin_, which is the binary used by the deployment engine to manage a resource. These plugins are stored in the _plugin cache_ (located in `~/.pulumi/plugins`) and can be manged with the `pulumi plugin` set of commands.
+2. An _SDK_ which provides bindings for each type of resource the provider can manage.
+
+Like the language runtime itself, the SDKs are available as regular packages.  For example, the for Node there is an `@pulumi/aws` package published to NPM and a `pulumi_aws` package for Python published to PyPI.  When these packages are added to your project, they run `pulumi plugin install` behind the scenes in order to download the resource plugin from pulumi.com.
```

Comment:
Can't suggest for the entire paragraph but please change to:

Like the language runtime itself, the SDKs are available as regular packages.  For example, there is a `@pulumi/aws` package for Node published to NPM and a `pulumi_aws` package for Python published to PyPI.  When these packages are added to your project, they run `pulumi plugin install` behind the scenes in order to download the resource plugin from pulumi.com.